### PR TITLE
Mention regex in Readme

### PR DESCRIPTION
**Pull request by pinx**
Fixes https://github.com/whatyouhide/corsica/issues/14

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,6 +68,11 @@ defmodule MyApp.Endpoint do
 end
 ```
 
+Wildcards can be used in the origins specification, by using regex:
```
Comment (whatyouhide) [inline README.md]: I'd rephrase this as "Regular expressions can be used in the `:origins` list as well in order to match hosts based on a pattern"

Comment (whatyouhide) [inline README.md]: Also, blank line before the code block :bow

File: README.md [hunk 2]
```diff
@@ -68,6 +68,11 @@ defmodule MyApp.Endpoint do
...
+```elixir
+    origins: [~r{//foo.com}, ~r{//localhost}]
```
Comment (whatyouhide) [inline README.md]: I'd also make the example more complete, e.g.:

```elixir
plug Corsica, origins: [~r{//foo\.com}, ~r{//localhost}]
```

File: README.md (hunk 1 above)
Comment (pinx) [inline README.md]: I intentionally started the line with "Wildcards", as a kind of header of the paragraph. Wildcards is what you want to achieve, regex is the means to this end, that was my line of thought. Should I give the paragraph it's own header, and use your phrases?

File: README.md (hunk 2 above)
Comment (pinx) [inline README.md]: I realized later, that I should match at the beginning, so this is the better way: ^https?:\/\/foo\.com

Comment (whatyouhide) [inline README.md]: I think `~r{//foo\.com$}` is enough, but your version is fine as well.

Comment (whatyouhide) [inline README.md]: Actually, maybe let's make the regex useful (because in the case above, `"http://foo.com"` would likely do the work as well :) Something like `~r{[a-zA-Z\-_]*\.example\.com}` to match all the subdomains of `example.com`. Wdyt?

**Comment by whatyouhide** [comment]
Ok, I thought about this a bit and I think this use case doesn't deserve a section in the README. The ["Overview"](https://github.com/whatyouhide/corsica#overview) section in the README is purposefully kept small, as it's just an overview over the functionalities of Corsica; the actual documentation belongs to hexdocs.pm :).

That said, I agree that this is a good feature in Corsica and should be promoted more. What do you think about making this more obvious in the *examples* in the README? The current example:

```elixir
defmodule MyApp.CORS do
  use Corsica.Router,
    origins: ["http://foo.com", "http://bar.com"],
    allow_credentials: true,
    max_age: 600,

  resource "/public/*", origins: "*"
  resource "/*"
end
```

could become something like 

```elixir
defmodule MyApp.CORS do
  use Corsica.Router,
    origins: ~r{[a-zA-Z\-_]*\.example\.com},
    allow_credentials: true,
    max_age: 600,

  resource "/public/*", origins: "*"
  resource "/*"
end
```

Still not 100% sure of this solution, but if we go with your solution (section in the README), then we should make the README more verbose and add more examples/feature. Wdyt?

**Comment by pinx** [comment]
I think you are right, in that we need to keep the Readme as short as possible. The typical values of origins is either * (I don't care) or ~r{^https?:\/\/(.*\.?)foo\.com} (I don't care about http or https, host name, or port; domain is important, and make sure it's at the beginning). (I'm not sure if the ^ is really necessary; probably anything after the host address is already removed by the browser, but maybe to prevent people forging funny request headers, with my host's address hidden at the end; I found it in an example).
So, you could include the regex I propose here, that is good for most users.
Also, you could expand the landing page in the hexdoc (https://hexdocs.pm/corsica/overview.html), to explain more on the "origin" subject. I think for first-time users, who just want to get their SPA working, it's enough to set the origin right.

**Comment by whatyouhide** [comment]
@pinx ok, so let's keep the README short and add `~r{^https?://(.*\.?)foo\.com}` to the example as I showed above. Then, let's expand the "Origin" section in the documentation for the `Corsica` module (I'll make sure https://hexdocs.pm/corsica/overview.html points to the `Corsica` module). 

Do you want to work on this changes (you could `push --force` over this PR) or do you prefer I tackle this? Anyways before I forget, thanks so much for the contribution :heart_decoration:

**Comment by pinx** [comment]
The cleanest way would be to clear our traces, and you make the changes.
The discussion was good, the changes are small.
Thanks for your contribution to the community.

Op 18:35 vr 25 mrt. 2016 schreef Andrea Leopardi <notifications@github.com>:

**Comment by whatyouhide** [comment]
@pinx the discussion was definitely helpful, thanks so much for it! I'll mention you in the commits that will come out of this then :)
